core/src/main/scala/cats/data/Stream.scala

```diff
@@ -0,0 +1,713 @@
+package cats
+package data
+
+import cats.syntax.order._
+import scala.reflect.ClassTag
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+sealed abstract class Stream[A] { lhs =>
```

Comment:
Do we want to possible name this something else to avoid clashing with the std lib, which I think has `Stream` imported by default?